--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -318,6 +318,7 @@ IF(LIBUSB_FOUND)
   MESSAGE(STATUS "LIBUSB found at: ${LIBUSB_LIBRARIES}")
   add_definitions(-DWITH_LIBUSB)
   target_link_libraries(domoticz ${LIBUSB_LIBRARIES})
+  target_link_libraries(domoticz usb-1.0) #is required for openwrt due to libusb-compat
 else()
   MESSAGE(STATUS "==== LibUSB not found, support for TE923/Voltcraft disabled!")
 ENDIF(LIBUSB_FOUND)
