Index: foris-pakon-plugin/src/static/sass/pakon.sass
===================================================================
--- foris-pakon-plugin.orig/src/static/sass/pakon.sass
+++ foris-pakon-plugin/src/static/sass/pakon.sass
@@ -153,7 +153,7 @@ dialog.loading
   visibility: hidden
   display: block
   position: fixed
-  background: url(/foris/config/static/img/loader.gif)
+  background: url(../../../img/loader.gif)
   width: 120px
   height: 120px
   padding: 0
@@ -165,4 +165,4 @@ dialog.loading
   &[open]
     visibility: visible
     top: 50%
-    opacity: 1
\ No newline at end of file
+    opacity: 1
