Index: ludus-0.6/ludus.py
===================================================================
--- ludus-0.6.orig/ludus.py
+++ ludus-0.6/ludus.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!/usr/bin/python3
 #  Copyright (C) 2017  Sebastian Garcia, Ondrej Lukas
 #
 #  This program is free software; you can redistribute it and/or modify
@@ -24,6 +24,7 @@
 #TODO:
 #   changes in suricata.yaml!
 #   check if sentinel is running !
+import configparser
 import time,datetime
 import sys
 import subprocess
