Index: python-jsonschema-2.6.0/setup.py
===================================================================
--- python-jsonschema-2.6.0.orig/setup.py
+++ python-jsonschema-2.6.0/setup.py
@@ -29,7 +29,6 @@ setup(
     name="jsonschema",
     packages=["jsonschema", "jsonschema.tests"],
     package_data={"jsonschema": ["schemas/*.json"]},
-    setup_requires=["vcversioner>=2.16.0.0"],
     extras_require=extras_require,
     author="Julian Berman",
     author_email="Julian@GrayVines.com",
@@ -38,6 +37,6 @@ setup(
     license="MIT",
     long_description=long_description,
     url="http://github.com/Julian/jsonschema",
+    version=os.environ.get('PKG_VERSION'),
     entry_points={"console_scripts": ["jsonschema = jsonschema.cli:main"]},
-    vcversioner={"version_module_paths" : ["jsonschema/_version.py"]},
 )
