From c77a02bc0f0d7a7c8eb70eb7efe38d001e677dbd Mon Sep 17 00:00:00 2001
From: Michal Hrusecky <michal.hrusecky@turris.com>
Date: Wed, 22 Dec 2021 10:44:23 +0100
Subject: [PATCH 5/6] Fix sync command

---
 schnapps.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/schnapps.sh b/schnapps.sh
index ceef300..2cf9765 100755
--- a/schnapps.sh
+++ b/schnapps.sh
@@ -716,7 +716,7 @@ upload() {
 }
 
 info2num() {
-    echo "$1" | sed -n 's|.*[/-]\([0-9]*\).info$|\1|'
+    echo "$1" | sed -n 's|.*[/-]\([0-9]*\).info$|\1|p'
 }
 
 sync_snps() {
@@ -732,14 +732,14 @@ sync_snps() {
         local num="$(info2num "$info")"
         if expr "$SYNC_TYPES" : ".*$TYPE.*" > /dev/null; then
             [ -f "$TMP_RMT_MNT_DIR/$REMOTE_PATH/$BOARD-medkit-$HOSTNAME-$num.info" ] \
-                || upload "$i"
+                || upload "$num"
         fi
     done
     if [ "$REMOTE_KEEP" != 1 ]; then
         for info in "$TMP_RMT_MNT_DIR/$REMOTE_PATH/$BOARD-medkit-$HOSTNAME-"*.info; do
             [ -f "$info" ] || continue
             local num="$(info2num "$info")"
-            [ -f "$TMP_MNT_DIR/$num.info" ] || continue
+            [ \! -f "$TMP_MNT_DIR/$num.info" ] || continue
             rm -f "$TMP_RMT_MNT_DIR/$REMOTE_PATH/$BOARD-medkit-$HOSTNAME-$num.info" \
                   "$TMP_RMT_MNT_DIR/$REMOTE_PATH/$BOARD-medkit-$HOSTNAME-$num.tar.gz" \
                   "$TMP_RMT_MNT_DIR/$REMOTE_PATH/$BOARD-medkit-$HOSTNAME-$num.tar.gz.pgp"
-- 
2.34.1

