--- a/setup.py
+++ b/setup.py
@@ -1,4 +1,10 @@
 from setuptools import setup
+from setuptools.command.install import install as InstallCommand
+
+class DependencyInstall(InstallCommand):
+    def run(self):
+	# Install requirements
+        self.do_egg_install()

 if __name__ == "__main__":

@@ -27,7 +33,6 @@ if __name__ == "__main__":
         use_incremental=True,
         install_requires=[
             "six",
-            "Twisted>=13.2",
             "werkzeug",
             "incremental",
         ],
@@ -39,4 +44,5 @@ if __name__ == "__main__":
         url="https://github.com/twisted/klein",
         maintainer='Amber Brown (HawkOwl)',
         maintainer_email='hawkowl@twistedmatrix.com',
+	cmdclass={"install":DependencyInstall },
     )
