--- a/unix/Makefile.in
+++ b/unix/Makefile.in
@@ -730,7 +730,7 @@
 	fi
 	@EXTRA_INSTALL_BINARIES@
 
-install-libraries: libraries $(INSTALL_TZDATA) install-msgs
+install-libraries: libraries  
 	@for i in "$(INCLUDE_INSTALL_DIR)" "$(SCRIPT_INSTALL_DIR)"; \
 	    do \
 	    if [ ! -d "$$i" ] ; then \
@@ -793,54 +793,6 @@
 	        "$(SCRIPT_INSTALL_DIR)"/tm.tcl; \
 	fi
 
-install-tzdata:
-	@for i in tzdata; \
-	    do \
-	    if [ ! -d "$(SCRIPT_INSTALL_DIR)"/$$i ] ; then \
-		echo "Making directory $(SCRIPT_INSTALL_DIR)/$$i"; \
-		$(INSTALL_DATA_DIR) "$(SCRIPT_INSTALL_DIR)"/$$i; \
-		else true; \
-		fi; \
-	    done;
-	@echo "Installing time zone files to $(SCRIPT_INSTALL_DIR)/tzdata/"
-	@for i in $(TOP_DIR)/library/tzdata/* ; do \
-	    if [ -d $$i ] ; then \
-		ii=`basename $$i`; \
-		if [ ! -d "$(SCRIPT_INSTALL_DIR)"/tzdata/$$ii ] ; then \
-		    $(INSTALL_DATA_DIR) "$(SCRIPT_INSTALL_DIR)"/tzdata/$$ii; \
-		fi; \
-		for j in $$i/* ; do \
-		    if [ -d $$j ] ; then \
-			jj=`basename $$j`; \
-			if [ ! -d "$(SCRIPT_INSTALL_DIR)"/tzdata/$$ii/$$jj ] ; then \
-			    $(INSTALL_DATA_DIR) "$(SCRIPT_INSTALL_DIR)"/tzdata/$$ii/$$jj; \
-			fi; \
-			for k in $$j/* ; do \
-			    $(INSTALL_DATA) $$k "$(SCRIPT_INSTALL_DIR)"/tzdata/$$ii/$$jj; \
-			done; \
-		    else \
-			$(INSTALL_DATA) $$j "$(SCRIPT_INSTALL_DIR)"/tzdata/$$ii; \
-		    fi; \
-		done; \
-	    else \
-		$(INSTALL_DATA) $$i "$(SCRIPT_INSTALL_DIR)"/tzdata; \
-	    fi; \
-	done;
-
-install-msgs:
-	@for i in msgs; \
-	    do \
-	    if [ ! -d "$(SCRIPT_INSTALL_DIR)"/$$i ] ; then \
-		echo "Making directory $(SCRIPT_INSTALL_DIR)/$$i"; \
-		$(INSTALL_DATA_DIR) "$(SCRIPT_INSTALL_DIR)"/$$i; \
-		else true; \
-		fi; \
-	    done;
-	@echo "Installing message catalog files to $(SCRIPT_INSTALL_DIR)/msgs/"
-	@for i in $(TOP_DIR)/library/msgs/*.msg ; do \
-		$(INSTALL_DATA) $$i "$(SCRIPT_INSTALL_DIR)"/msgs; \
-	done;
-
 install-doc: doc
 	@for i in "$(MAN_INSTALL_DIR)" "$(MAN1_INSTALL_DIR)" "$(MAN3_INSTALL_DIR)" "$(MANN_INSTALL_DIR)" ; \
 	    do \
@@ -1665,12 +1617,6 @@
 	    done;
 	mkdir $(DISTDIR)/library/encoding
 	cp -p $(TOP_DIR)/library/encoding/*.enc $(DISTDIR)/library/encoding
-	mkdir $(DISTDIR)/library/msgs
-	cp -p $(TOP_DIR)/library/msgs/*.msg $(DISTDIR)/library/msgs
-	( cd $(TOP_DIR); \
-	  find library/tzdata -name CVS -prune -o -type f -print ) \
-	    | ( cd $(TOP_DIR) ; xargs tar cf - ) \
-	    | ( cd $(DISTDIR) ; tar xfp - )
 	mkdir $(DISTDIR)/doc
 	cp -p $(TOP_DIR)/license.terms $(TOP_DIR)/doc/*.[13n] \
 		$(TOP_DIR)/doc/man.macros $(DISTDIR)/doc
