From 6cf0bf894218c2444a85a2e0725274002147f330 Mon Sep 17 00:00:00 2001
From: Pavel Spirek <pavel.spirek@nic.cz>
Date: Wed, 4 Nov 2015 08:44:58 +0100
Subject: [PATCH] Disable infrared

---
 pifacecad/__init__.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pifacecad/__init__.py b/pifacecad/__init__.py
index 2c9b646..b41b0c4 100644
--- a/pifacecad/__init__.py
+++ b/pifacecad/__init__.py
@@ -41,6 +41,6 @@ from .lcd import (
     LCDBitmap,
 )
 
-from .ir import (
-    IREventListener,
-)
+#from .ir import (
+#    IREventListener,
+#)
-- 
1.9.1

