diff --git a/src/flow.c b/src/flow.c
index c371b54e..20bb4029 100644
--- a/src/flow.c
+++ b/src/flow.c
@@ -309,6 +309,14 @@ void FlowHandlePacketUpdate(Flow *f, Packet *p)
             f->flags &= ~FLOW_PROTO_DETECT_TS_DONE;
             p->flags |= PKT_PROTO_DETECT_TS_DONE;
         }
+#ifdef NFQ
+    if (f->if_src==0) f->if_src=p->nfq_v.ifi;
+    if (!f->hw_address_src_known && p->nfq_v.hw_address_known){
+        SCLogDebug("setting src MAC address %02x:%02x:%02x:%02x:%02x:%02x", p->nfq_v.hw_address[0], p->nfq_v.hw_address[1], p->nfq_v.hw_address[2], p->nfq_v.hw_address[3], p->nfq_v.hw_address[4], p->nfq_v.hw_address[5]);
+        f->hw_address_src_known = 1;
+        memcpy(f->hw_address_src, p->nfq_v.hw_address, 6);
+    }
+#endif
     } else {
         f->tosrcpktcnt++;
         f->tosrcbytecnt += GET_PKT_LEN(p);
@@ -324,6 +331,14 @@ void FlowHandlePacketUpdate(Flow *f, Packet *p)
             f->flags &= ~FLOW_PROTO_DETECT_TC_DONE;
             p->flags |= PKT_PROTO_DETECT_TC_DONE;
         }
+#ifdef NFQ
+    if (f->if_dst==0) f->if_dst=p->nfq_v.ifo;
+    if (!f->hw_address_dst_known && p->nfq_v.hw_address_known){
+        SCLogDebug("setting dst MAC address %02x:%02x:%02x:%02x:%02x:%02x", p->nfq_v.hw_address[0], p->nfq_v.hw_address[1], p->nfq_v.hw_address[2], p->nfq_v.hw_address[3], p->nfq_v.hw_address[4], p->nfq_v.hw_address[5]);
+        f->hw_address_dst_known = 1;
+        memcpy(f->hw_address_dst, p->nfq_v.hw_address, 6);
+    }
+#endif
     }
 
     if ((f->flags & (FLOW_TO_DST_SEEN|FLOW_TO_SRC_SEEN)) == (FLOW_TO_DST_SEEN|FLOW_TO_SRC_SEEN)) {
diff --git a/src/flow.h b/src/flow.h
index 93ca7afc..fe696676 100644
--- a/src/flow.h
+++ b/src/flow.h
@@ -435,6 +435,14 @@ typedef struct Flow_
     uint32_t tosrcpktcnt;
     uint64_t todstbytecnt;
     uint64_t tosrcbytecnt;
+#ifdef NFQ
+    uint8_t hw_address_src[6];
+    uint8_t hw_address_src_known;
+    uint8_t hw_address_dst[6];
+    uint8_t hw_address_dst_known;
+    uint32_t if_src;
+    uint32_t if_dst;
+#endif
 } Flow;
 
 enum FlowState {
diff --git a/src/output-json-flow.c b/src/output-json-flow.c
index f715cc04..69b5c15e 100644
--- a/src/output-json-flow.c
+++ b/src/output-json-flow.c
@@ -174,6 +174,42 @@ static json_t *CreateJSONHeaderFromFlow(Flow *f, const char *event_type)
                     json_integer(f->code));
             break;
     }
+
+#ifdef NFQ
+      // flow doesn't know ether header - unless it's our hacked NFQ
+      if  (f->hw_address_src_known || f->hw_address_dst_known) {
+        json_t *ejs = json_object();
+        if (ejs != NULL) {
+            char eth_addr[19];
+            const uint8_t *src=NULL, *dst=NULL;
+
+            if (f->hw_address_src_known) src=f->hw_address_src;
+            if (f->hw_address_dst_known) dst=f->hw_address_dst;
+            if (src) {
+                snprintf(eth_addr, 19, "%02x:%02x:%02x:%02x:%02x:%02x",
+                        src[0],
+                        src[1],
+                        src[2],
+                        src[3],
+                        src[4],
+                        src[5]);
+                json_object_set_new(ejs, "src", json_string(eth_addr));
+            }
+            if (dst) {
+                snprintf(eth_addr, 19, "%02x:%02x:%02x:%02x:%02x:%02x",
+                        dst[0],
+                        dst[1],
+                        dst[2],
+                        dst[3],
+                        dst[4],
+                        dst[5]);
+                json_object_set_new(ejs, "dst", json_string(eth_addr));
+            }
+            json_object_set_new(js, "ether", ejs);
+        }
+    }
+#endif
+
     return js;
 }
 
diff --git a/src/output-json.c b/src/output-json.c
index 32495d41..d6850140 100644
--- a/src/output-json.c
+++ b/src/output-json.c
@@ -470,6 +470,50 @@ json_t *CreateJSONHeader(const Packet *p, int direction_sensitive,
             break;
     }
 
+    if (p->ethh
+#ifdef NFQ
+        || (p->flow && (p->flow->hw_address_src_known || p->flow->hw_address_dst_known))
+#endif
+    ) {
+        json_t *ejs = json_object();
+        if (ejs != NULL) {
+            char eth_addr[19];
+            const uint8_t *src=NULL, *dst=NULL;
+
+            if (p->ethh) {
+                src = p->ethh->eth_src;
+                dst = p->ethh->eth_dst;
+            }
+#ifdef NFQ
+            else {
+                if (p->flow->hw_address_src_known) src=p->flow->hw_address_src;
+                if (p->flow->hw_address_dst_known) dst=p->flow->hw_address_dst;
+            }
+#endif
+            if (src) {
+                snprintf(eth_addr, 19, "%02x:%02x:%02x:%02x:%02x:%02x",
+                        src[0],
+                        src[1],
+                        src[2],
+                        src[3],
+                        src[4],
+                        src[5]);
+                json_object_set_new(ejs, "src", json_string(eth_addr));
+            }
+            if (dst) {
+                snprintf(eth_addr, 19, "%02x:%02x:%02x:%02x:%02x:%02x",
+                        dst[0],
+                        dst[1],
+                        dst[2],
+                        dst[3],
+                        dst[4],
+                        dst[5]);
+                json_object_set_new(ejs, "dst", json_string(eth_addr));
+            }
+            json_object_set_new(js, "ether", ejs);
+        }
+    }
+
     return js;
 }
 
diff --git a/src/source-nfq.c b/src/source-nfq.c
index 4c13f5a7..0acb1099 100644
--- a/src/source-nfq.c
+++ b/src/source-nfq.c
@@ -426,6 +426,7 @@ static int NFQSetupPkt (Packet *p, struct nfq_q_handle *qh, void *data)
     int ret;
     char *pktdata;
     struct nfqnl_msg_packet_hdr *ph;
+    struct nfqnl_msg_packet_hw *hwaddr;
 
     ph = nfq_get_msg_packet_hdr(tb);
     if (ph != NULL) {
@@ -433,6 +434,13 @@ static int NFQSetupPkt (Packet *p, struct nfq_q_handle *qh, void *data)
         //p->nfq_v.hw_protocol = ntohs(p->nfq_v.ph->hw_protocol);
         p->nfq_v.hw_protocol = ph->hw_protocol;
     }
+    hwaddr = nfq_get_packet_hw(tb);
+    if (hwaddr != NULL) {
+        p->nfq_v.hw_address_known = 1;
+        memcpy(p->nfq_v.hw_address, hwaddr->hw_addr, 6);
+    } else {
+        p->nfq_v.hw_address_known = 0;
+    }
     /* coverity[missing_lock] */
     p->nfq_v.mark = nfq_get_nfmark(tb);
     if (nfq_config.mode == NFQ_REPEAT_MODE) {
diff --git a/src/source-nfq.h b/src/source-nfq.h
index 374b3332..98af68f1 100644
--- a/src/source-nfq.h
+++ b/src/source-nfq.h
@@ -45,6 +45,8 @@ typedef struct NFQPacketVars_
     uint32_t ifi;
     uint32_t ifo;
     uint16_t hw_protocol;
+    uint8_t hw_address_known;
+    uint8_t hw_address[6];
 } NFQPacketVars;
 
 typedef struct NFQQueueVars_
diff --git a/src/flow-util.h b/src/flow-util.h
index 374b3332..98af68f1 100644
--- a/src/flow-util.h
+++ b/src/flow-util.h
@@ -29,12 +29,23 @@
 
 #define COPY_TIMESTAMP(src,dst) ((dst)->tv_sec = (src)->tv_sec, (dst)->tv_usec = (src)->tv_usec)
 
+#ifdef NFQ
 #define RESET_COUNTERS(f) do { \
         (f)->todstpktcnt = 0; \
         (f)->tosrcpktcnt = 0; \
         (f)->todstbytecnt = 0; \
         (f)->tosrcbytecnt = 0; \
+       (f)->hw_address_src_known = 0; \
+       (f)->hw_address_dst_known = 0; \
     } while (0)
+#else
+#define RESET_COUNTERS(f) do { \
+        (f)->todstpktcnt = 0; \
+        (f)->tosrcpktcnt = 0; \
+        (f)->todstbytecnt = 0; \
+        (f)->tosrcbytecnt = 0; \
+    } while (0)
+#endif
 
 #define FLOW_INITIALIZE(f) do { \
         (f)->sp = 0; \
