--- a/protolib/unix/Makefile.linux
+++ b/protolib/unix/Makefile.linux
@@ -39,7 +39,7 @@ SYSTEM_LIBS = -ldl -lrt
 # (We export these for other Makefiles as needed)
 #
 
-SYSTEM_HAVES = -DLINUX -DHAVE_IPV6 -DHAVE_GETLOGIN -D_FILE_OFFSET_BITS=64 -DHAVE_LOCKF \
+SYSTEM_HAVES = -DLINUX -DHAVE_GETLOGIN -D_FILE_OFFSET_BITS=64 -DHAVE_LOCKF \
 -DHAVE_OLD_SIGNALHANDLER -DHAVE_DIRFD -DHAVE_ASSERT -DNO_SCM_RIGHTS -DHAVE_SCHED
 
 # (TBD) Move ProtoRouteMgr to ProtokitEx ??
--- a/unix/Makefile.linux
+++ b/unix/Makefile.linux
@@ -35,7 +35,7 @@ SYSTEM_LIBS = -ldl -lpthread
 # (We export these for other Makefiles as needed)
 #
 
-SYSTEM_HAVES = -DLINUX -DHAVE_PSELECT -DHAVE_SCHED -DHAVE_IPV6 -DHAVE_GETLOGIN -DHAVE_LOCKF -DHAVE_DIRFD \
+SYSTEM_HAVES = -DLINUX -DHAVE_PSELECT -DHAVE_SCHED -DHAVE_GETLOGIN -DHAVE_LOCKF -DHAVE_DIRFD \
 -DHAVE_ASSERT $(NETSEC)
 
 SYSTEM = linux
