--- a/setup.py
+++ b/setup.py
@@ -6,8 +6,8 @@ from os.path import abspath, dirname, jo
 from setuptools import setup
 
 PROJECT_ROOT = abspath(dirname(__file__))
-with open(join(PROJECT_ROOT, 'README.rst')) as f:
-    readme = f.read()
+# with open(join(PROJECT_ROOT, 'README.rst')) as f:
+#     readme = f.read()
 
 version = (
     [l for l in open(join(PROJECT_ROOT, 'zeroconf.py')) if '__version__' in l][0]
@@ -20,7 +20,7 @@ setup(
     version=version,
     description='Pure Python Multicast DNS Service Discovery Library '
     '(Bonjour/Avahi compatible)',
-    long_description=readme,
+    # long_description=readme,
     author='Paul Scott-Murphy, William McBrine, Jakub Stasiak',
     url='https://github.com/jstasiak/python-zeroconf',
     py_modules=['zeroconf'],
