diff --git a/src/rmbt_flow.h b/src/rmbt_flow.h
index 310eb29..cba5814 100644
--- a/src/rmbt_flow.h
+++ b/src/rmbt_flow.h
@@ -20,8 +20,6 @@
 #include "rmbt_common.h"
 #include "rmbt_result.h"
 
-#include <stdatomic.h>
-
 typedef struct {
 	char *bind_ip, *server_host, *server_port, *cipherlist, *secret, *token, *test_id, *file_summary, *file_flows, *file_stats;
 	int_fast16_t dl_num_flows, ul_num_flows, dl_duration_s, ul_duration_s, rtt_tcp_payload_num, dl_pretest_duration_s, ul_pretest_duration_s, dl_wait_time_s,
@@ -32,7 +30,7 @@ typedef struct {
 } TestConfig;
 
 typedef struct {
-	atomic_bool global_abort;
+	volatile bool global_abort;
 	pthread_mutex_t mutex;
 	pthread_cond_t cond;
 	int_fast16_t total;
