--- a/Makefile
+++ b/Makefile
@@ -18,23 +18,23 @@ endif
 $(eval $(call find_lib,libknot))
 $(eval $(call find_lib,libuv,1.0))
 $(eval $(call find_alt,lua,luajit))
-$(eval $(call find_lib,cmocka))
-$(eval $(call find_bin,doxygen))
-$(eval $(call find_bin,sphinx-build))
-$(eval $(call find_lib,libmemcached,1.0))
-$(eval $(call find_lib,hiredis))
+#$(eval $(call find_lib,cmocka))
+#$(eval $(call find_bin,doxygen))
+#$(eval $(call find_bin,sphinx-build))
+#$(eval $(call find_lib,libmemcached,1.0))
+#$(eval $(call find_lib,hiredis))
 $(eval $(call find_lib,socket_wrapper))
 $(eval $(call find_lib,libdnssec))
-$(eval $(call find_gopkg,geoip,github.com/abh/geoip))
+#$(eval $(call find_gopkg,geoip,github.com/abh/geoip))
 # Find Go compiler version
 # @note Go 1.5 support amd64 only, disabled for other architectures
-ifeq ($(ARCH),x86_64)
-E :=
-GO_VERSION := $(subst $(E) $(E),,$(subst go,,$(wordlist 1,2,$(subst ., ,$(word 3,$(shell $(GO) version))))))
-$(eval $(call find_ver,go,$(GO_VERSION),15))
-else
-HAS_go = no
-endif
+#ifeq ($(ARCH),x86_64)
+#E :=
+#GO_VERSION := $(subst $(E) $(E),,$(subst go,,$(wordlist 1,2,$(subst ., ,$(word 3,$(shell $(GO) version))))))
+#$(eval $(call find_ver,go,$(GO_VERSION),15))
+#else
+#HAS_go = no
+#endif
 # Work around luajit on OS X
 ifeq ($(PLATFORM), Darwin)
 ifneq (,$(findstring luajit, $(lua_LIBS)))
@@ -66,15 +66,15 @@ info:
 	$(info [$(HAS_lua)] LuaJIT (daemon))
 	$(info [$(HAS_libuv)] libuv (daemon))
 	$(info )
-	$(info Optional)
-	$(info --------)
-	$(info [$(HAS_doxygen)] doxygen (doc))
-	$(info [$(HAS_go)] Go 1.5+ on amd64 (modules/go))
-	$(info [$(HAS_geoip)] github.com/abh/geoip (modules/tinyweb))
-	$(info [$(HAS_libmemcached)] libmemcached (modules/memcached))
-	$(info [$(HAS_hiredis)] hiredis (modules/redis))
-	$(info [$(HAS_cmocka)] cmocka (tests/unit))
-	$(info )
+#	$(info Optional)
+#	$(info --------)
+#	$(info [$(HAS_doxygen)] doxygen (doc))
+#	$(info [$(HAS_go)] Go 1.5+ on amd64 (modules/go))
+#	$(info [$(HAS_geoip)] github.com/abh/geoip (modules/tinyweb))
+#	$(info [$(HAS_libmemcached)] libmemcached (modules/memcached))
+#	$(info [$(HAS_hiredis)] hiredis (modules/redis))
+#	$(info [$(HAS_cmocka)] cmocka (tests/unit))
+#	$(info )
 
 # Installation directories
 $(DESTDIR)$(MODULEDIR):
