diff --git a/lib/defines.h b/lib/defines.h
index a25d565..c70a651 100644
--- a/lib/defines.h
+++ b/lib/defines.h
@@ -56,7 +56,7 @@ static inline int __attribute__((__cold__)) kr_error(int x) {
  * @cond internal
  */
 #define KR_CONN_RTT_MAX 3000 /* Timeout for network activity */
-#define KR_CONN_RETRY 250    /* Retry interval for network activity */
+#define KR_CONN_RETRY 100    /* Retry interval for network activity */
 #define KR_ITER_LIMIT 100    /* Built-in iterator limit */
 #define KR_CNAME_CHAIN_LIMIT 40 /* Built-in maximum CNAME chain length */
 #define KR_TIMEOUT_LIMIT 4   /* Maximum number of retries after timeout. */
