From dcc75a1ee27d3ed340d88599fb80a7ffb3184387 Mon Sep 17 00:00:00 2001
From: Grigorii Demidov <grigorii.demidov@nic.cz>
Date: Fri, 21 Apr 2017 14:30:05 +0200
Subject: [PATCH] lib/nsrep: revert some changes from commit 5581cf9b

---
 lib/nsrep.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/lib/nsrep.c b/lib/nsrep.c
index 1bf2704..8139a22 100644
--- a/lib/nsrep.c
+++ b/lib/nsrep.c
@@ -139,16 +139,16 @@ static int eval_nsrep(const char *k, void *v, void *baton)
 		/* If the server doesn't have IPv6, give it disadvantage. */
 		if (reputation & KR_NS_NOIP6) {
 			score += FAVOUR_IPV6;
+			/* If the server is unknown but has rep record, treat it as timeouted */
 			if (reputation & KR_NS_NOIP4) {
-				/* Server is unknown but has rep record.
-				 * We can not distinguish if it happens either
-				 * due to timeout or due to other circumstances
-				 * (for example, we have ipv6-only network and
-				 * we are dealing with ipv4-only NS).
-				 * Don't use it for now.
-				 * TODO -
-				 * add explicit flag for timeouted servers */
-				score = KR_NS_MAX_SCORE + 1;
+				score = KR_NS_UNKNOWN;
+				/* Try to start with clean slate */
+				if (!(ctx->options & QUERY_NO_IPV6)) {
+					reputation &= ~KR_NS_NOIP6;
+				}
+				if (!(ctx->options & QUERY_NO_IPV4)) {
+					reputation &= ~KR_NS_NOIP4;
+				}
 			}
 		}
 	} else {
--
libgit2 0.25.0

