--- a/Makefile
+++ b/Makefile
@@ -61,11 +61,11 @@ BUILD_CFLAGS += $(libknot_CFLAGS) $(libu
 BUILD_CFLAGS += $(addprefix -I,$(wildcard contrib/ccan/*) contrib/murmurhash3)
 
 # Check if it has libknot 2.3.0 and nettle to support DNS cookies
-$(eval $(call find_alt,knot230,libknot,2.3))
-ifeq ($(HAS_nettle)|$(HAS_knot230),yes|yes)
-BUILD_CFLAGS += -DENABLE_COOKIES
-ENABLE_COOKIES := yes
-endif
+# $(eval $(call find_alt,knot230,libknot,2.3))
+# ifeq ($(HAS_nettle)|$(HAS_knot230),yes|yes)
+# BUILD_CFLAGS += -DENABLE_COOKIES
+# ENABLE_COOKIES := yes
+# endif
 
 # Overview
 info:
