--- a/lib/defines.h
+++ b/lib/defines.h
@@ -51,7 +51,7 @@ static inline int __attribute__((__cold_
  * @cond internal
  */
 #define KR_CONN_RTT_MAX 3000 /* Timeout for network activity */
-#define KR_CONN_RETRY 300    /* Retry interval for network activity */
+#define KR_CONN_RETRY 100    /* Retry interval for network activity */
 #define KR_ITER_LIMIT 50     /* Built-in iterator limit */
 #define KR_CNAME_CHAIN_LIMIT 40 /* Built-in maximum CNAME chain length */
 #define KR_TIMEOUT_LIMIT 4   /* Maximum number of retries after timeout. */
