Index: python3-foris-data_collect-plugin-1.4/setup.py
===================================================================
--- python3-foris-data_collect-plugin-1.4.orig/setup.py
+++ python3-foris-data_collect-plugin-1.4/setup.py
@@ -30,10 +30,6 @@ setup(
         "foris @ git+https://gitlab.labs.nic.cz/turris/foris.git#egg=foris",
     ],
     setup_requires=[
-        'babel',
-        'jinja2',
-        'libsass',
-        'foris-plugins-distutils',
     ],
     provides=[
         "foris_plugins.data_collect",
