diff -u -r socat-1.7.3.0/socat.c socat-1.7.3.0b/socat.c
--- socat-1.7.3.0/socat.c	2015-01-24 11:15:22.000000000 +0100
+++ socat-1.7.3.0b/socat.c	2015-07-03 13:24:23.326489458 +0200
@@ -298,7 +298,6 @@
       Sigaction(SIGABRT, &act, NULL);
       Sigaction(SIGBUS,  &act, NULL);
       Sigaction(SIGFPE,  &act, NULL);
-      Sigaction(SIGSEGV, &act, NULL);
       Sigaction(SIGTERM, &act, NULL);
    }
    Signal(SIGPIPE, SIG_IGN);
