--- a/net_io.c
+++ b/net_io.c
@@ -1705,5 +1705,6 @@ static void writeFATSVEventMessage(struc
     }
     p += snprintf(p, bufsize(p, end), "\t%s\t", datafield);
-    for (size_t i = 0; i < len; ++i) {
+    size_t i;
+    for (i = 0; i < len; ++i) {
         p += snprintf(p, bufsize(p, end), "%02X", data[i]);
     }
--- a/mode_s.c
+++ b/mode_s.c
@@ -755,7 +755,8 @@ static void decodeBDS20(struct modesMess
     // Catch possible bad decodings since BDS2,0 is not
     // 100% reliable: accept only alphanumeric data
     mm->callsign_valid = 1;
-    for (int i = 0; i < 8; ++i) {
+    int i;
+    for (i = 0; i < 8; ++i) {
         if (! ((mm->callsign[i] >= 'A' && mm->callsign[i] <= 'Z') ||
                (mm->callsign[i] >= '0' && mm->callsign[i] <= '9') ||
                mm->callsign[i] == ' ') ) {
