From 31cd99d4339127ef56554ca1b111993e9c03c0a2 Mon Sep 17 00:00:00 2001
From: Stepan Henek <stepan.henek@nic.cz>
Date: Mon, 12 Nov 2018 10:31:55 +0100
Subject: [PATCH] compatibility with python older than 3.6.5

---
 foris_ws/__main__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/foris_ws/__main__.py b/foris_ws/__main__.py
index 37bafcb..8e49353 100755
--- a/foris_ws/__main__.py
+++ b/foris_ws/__main__.py
@@ -51,7 +51,7 @@ except ModuleNotFoundError:
     pass
 
 
-def main() -> typing.NoReturn:
+def main():
     # Parse the command line options
     parser = argparse.ArgumentParser(prog="foris-ws")
     parser.add_argument("-d", "--debug", dest="debug", action="store_true", default=False)
-- 
2.19.2

