--- a/src/dvr/dvr_config.c
+++ b/src/dvr/dvr_config.c
@@ -177,8 +177,8 @@ dvr_config_create(const char *name, cons
   cfg->dvr_removal_days = DVR_RET_REM_FOREVER;
   cfg->dvr_clone = 1;
   cfg->dvr_tag_files = 1;
-  cfg->dvr_skip_commercials = 1;
-  dvr_charset_update(cfg, intlconv_filesystem_charset());
+  cfg->dvr_skip_commercials = 0;
+  dvr_charset_update(cfg, "UTF-8");
   cfg->dvr_warm_time = 30;
   cfg->dvr_update_window = 24 * 3600;
   cfg->dvr_pathname = strdup("$t$n.$x");
