diff --git a/src/Makefile b/src/Makefile
index 727794b..fe0f33d 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -31,10 +31,10 @@ LDFLAGS += $(MYLDFLAGS)
 all:
 
 install: $(CMOD) $(LMOD)
-	$(INSTALL) -d $(LUAPATH)/ssl $(LUACPATH)
-	$(INSTALL) $(CMOD) $(LUACPATH)
-	$(INSTALL) -m644 $(LMOD) $(LUAPATH)
-	$(INSTALL) -m644 https.lua $(LUAPATH)/ssl
+	$(INSTALL) -d $(DESTDIR)/$(LUAPATH)/ssl $(DESTDIR)/$(LUACPATH)
+	$(INSTALL) $(CMOD) $(DESTDIR)/$(LUACPATH)
+	$(INSTALL) -m644 $(LMOD) $(DESTDIR)/$(LUAPATH)
+	$(INSTALL) -m644 https.lua $(DESTDIR)/$(LUAPATH)/ssl
 
 linux:
 	@$(MAKE) $(CMOD) MYCFLAGS="$(LNX_CFLAGS)" MYLDFLAGS="$(LNX_LDFLAGS)" EXTRA="$(EXTRA)"
