--- a/tty.sh
+++ b/tty.sh
@@ -76,7 +76,7 @@
 #endif
 
 #include "config.h"
-#ifdef SVR4
+#if 0
 #include <sys/stropts.h>	/* for I_POP */
 #endif
 
--- a/screen.c
+++ b/screen.c
@@ -50,7 +50,7 @@
 
 #include "config.h"
 
-#ifdef SVR4
+#if 0
 # include <sys/stropts.h>
 #endif
 
--- a/process.c
+++ b/process.c
@@ -37,7 +37,7 @@
 #include "config.h"
 
 /* for solaris 2.1, Unixware (SVR4.2) and possibly others: */
-#ifdef SVR4
+#if 0
 # include <sys/stropts.h>
 #endif
 
