--- a/setup.py
+++ b/setup.py
@@ -4,7 +4,7 @@ from setuptools import setup, find_packa
 
 PACKAGE_NAME = 'homeassistant'
 HERE = os.path.abspath(os.path.dirname(__file__))
-with open(os.path.join(HERE, PACKAGE_NAME, 'const.py')) as fp:
+with open(os.path.join(HERE, PACKAGE_NAME, 'const.py'), encoding='utf-8') as fp:
     VERSION = re.search("__version__ = ['\"]([^']+)['\"]\n", fp.read()).group(1)
 DOWNLOAD_URL = \
     'https://github.com/balloob/home-assistant/archive/{}.zip'.format(VERSION)
