--- a/homeassistant/util/package.py
+++ b/homeassistant/util/package.py
@@ -7,6 +7,7 @@ import sys
 def install_package(package, upgrade=False, target=None):
     """Install a package on PyPi. Accepts pip compatible package strings.
     Return boolean if install successfull."""
+    return True
     # Not using 'import pip; pip.main([])' because it breaks the logger
     args = [sys.executable, '-m', 'pip', 'install', '--quiet', package]
     if upgrade:
