Index: alsa-lib-1.0.27.2/src/ucm/parser.c
===================================================================
--- alsa-lib-1.0.27.2.orig/src/ucm/parser.c
+++ alsa-lib-1.0.27.2/src/ucm/parser.c
@@ -33,6 +33,10 @@
 #include "ucm_local.h"
 #include <dirent.h>
 
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
 /** The name of the environment variable containing the UCM directory */
 #define ALSA_CONFIG_UCM_VAR "ALSA_CONFIG_UCM"
 
