--- a/src/lxc/lxc_snapshot.c
+++ b/src/lxc/lxc_snapshot.c
@@ -32,6 +32,8 @@
 #include "arguments.h"
 #include "utils.h"
 
+#include <../include/getline.h>
+
 lxc_log_define(lxc_snapshot, lxc);
 
 char *newname;
--- a/src/lxc/lxc_user_nic.c
+++ b/src/lxc/lxc_user_nic.c
@@ -49,6 +49,8 @@
 #include "config.h"
 #include "utils.h"
 
+#include <../include/getline.h>
+
 #if ISTEST
 #define CONF_FILE "/tmp/lxc-usernet"
 #define DB_FILE "/tmp/nics"
