--- a/Makefile
+++ b/Makefile
@@ -84,8 +84,8 @@ solaris:
 	$(MAKE) all CC="gcc -O -s" O=.o X= \
 	CFLAGS="-DUNIX" LFLAGS="" LIBS="-lsocket -lpthread -lnsl" OUT=-o
 linux:
-	$(MAKE) all CC="gcc -O -s" O=.o X= \
-	CFLAGS="-DUNIX" LFLAGS="" LIBS="-lpthread" OUT=-o
+	$(MAKE) all CC="$(CC) -O -s" O=.o X= \
+	LFLAGS="" LIBS="-lpthread" OUT=-o
 macosx:
 	$(MAKE) all CC="gcc -O" O=.o X= \
 	CFLAGS="-DUNIX -DSOCKLEN_T" LFLAGS="" LIBS="-lpthread" OUT=-o
