--- a/src/lj_arch.h
+++ b/src/lj_arch.h
@@ -328,10 +328,6 @@
 #if !(__ARM_EABI__ || LJ_TARGET_IOS)
 #error "Only ARM EABI or iOS 3.0+ ABI is supported"
 #endif
-#elif LJ_TARGET_PPC || LJ_TARGET_PPCSPE
-#if defined(_SOFT_FLOAT) || defined(_SOFT_DOUBLE)
-#error "No support for PowerPC CPUs without double-precision FPU"
-#endif
 #if defined(_LITTLE_ENDIAN)
 #error "No support for little-endian PowerPC"
 #endif
