--- a/include/msgpack/vrefbuffer.h	2017-08-04 17:02:07.000000000 +0200
+++ b/include/msgpack/vrefbuffer.h	2018-09-20 13:03:51.071078491 +0200
@@ -13,14 +13,7 @@
 #include "zone.h"
 #include <stdlib.h>
 
-#if defined(unix) || defined(__unix) || defined(__APPLE__) || defined(__OpenBSD__) || defined(__QNX__) || defined(__QNXTO__)
 #include <sys/uio.h>
-#else
-struct iovec {
-    void  *iov_base;
-    size_t iov_len;
-};
-#endif
 
 #ifdef __cplusplus
 extern "C" {
