--- a/parsetime.y
+++ b/parsetime.y
@@ -8,6 +8,9 @@
 
 #define YYDEBUG 1
 
+#define is_leap_year(y) \
+    ((y) % 4 == 0 && ((y) % 100 != 0 || (y) % 400 == 0))
+
 struct tm exectm;
 static int isgmt;
 static int yearspec;
@@ -217,8 +220,8 @@ date            : month_name day_number
 				 mnum == 12) && dnum > 31)
 			    || ((mnum ==  4 || mnum ==  6 || mnum ==  9 ||
 			         mnum == 11) && dnum > 30)
-			    || (mnum ==  2 && dnum > 29 &&  __isleap(ynum+1900))
-			    || (mnum ==  2 && dnum > 28 && !__isleap(ynum+1900))
+			    || (mnum ==  2 && dnum > 29 &&  is_leap_year(ynum+1900))
+			    || (mnum ==  2 && dnum > 28 && !is_leap_year(ynum+1900))
 			   )
 			{
 			    yyerror("Error in day of month");
@@ -261,8 +264,8 @@ date            : month_name day_number
 				 mnum == 12) && dnum > 31)
 			    || ((mnum ==  4 || mnum ==  6 || mnum ==  9 ||
 			         mnum == 11) && dnum > 30)
-			    || (mnum ==  2 && dnum > 29 &&  __isleap(ynum+1900))
-			    || (mnum ==  2 && dnum > 28 && !__isleap(ynum+1900))
+			    || (mnum ==  2 && dnum > 29 &&  is_leap_year(ynum+1900))
+			    || (mnum ==  2 && dnum > 28 && !is_leap_year(ynum+1900))
 			   )
 			{
 			    yyerror("Error in day of month");
