Index: w_scan-20170107/Makefile.in
===================================================================
--- w_scan-20170107.orig/Makefile.in
+++ w_scan-20170107/Makefile.in
@@ -836,7 +836,7 @@ version:
 	@echo "/* AUTOMATICALLY GENERATED - DO NOT EDIT MANUALLY */" > version.h
 	@echo "#ifndef W_SCAN_VERSION_H" >> version.h
 	@echo "#define W_SCAN_VERSION_H" >> version.h
-	@echo "uint version=$(__VERSION);" >> version.h
+	@echo "unsigned int version=$(__VERSION);" >> version.h
 	@echo "#endif" >> version.h
 	@echo "" >> version.h
 	@chmod a-x version.h
