Index: u-boot-mox-2018-11-rc1/u-boot/cmd/pxe.c
===================================================================
--- u-boot-mox-2018-11-rc1.orig/u-boot/cmd/pxe.c
+++ u-boot-mox-2018-11-rc1/u-boot/cmd/pxe.c
@@ -21,6 +21,9 @@
 #define MAX_TFTP_PATH_LEN 127
 
 const char *pxe_default_paths[] = {
+#ifdef CONFIG_SYS_BOARD
+	"default-" CONFIG_SYS_ARCH "-" CONFIG_SYS_SOC "-" CONFIG_SYS_BOARD,
+#endif
 #ifdef CONFIG_SYS_SOC
 	"default-" CONFIG_SYS_ARCH "-" CONFIG_SYS_SOC,
 #endif
Index: u-boot-mox-2018-11-rc1/u-boot/configs/turris_mox_defconfig
===================================================================
--- u-boot-mox-2018-11-rc1.orig/u-boot/configs/turris_mox_defconfig
+++ u-boot-mox-2018-11-rc1/u-boot/configs/turris_mox_defconfig
@@ -84,3 +84,14 @@ CONFIG_WDT=y
 CONFIG_WDT_ARMADA_37XX=y
 CONFIG_SHA1=y
 CONFIG_SHA256=y
+CONFIG_BOOTCOMMAND="run mox_boot"
+CONFIG_FIT=y
+CONFIG_FIT_ENABLE_SHA256_SUPPORT=y
+CONFIG_FIT_SIGNATURE=n
+CONFIG_FIT_VERBOSE=y
+CONFIG_FIT_BEST_MATCH=n
+CONFIG_CMD_LZMADEC=y
+CONFIG_CMD_AES=y
+CONFIG_CMD_HASH=y
+CONFIG_CMD_SHA1SUM=y
+CONFIG_NET_RANDOM_ETHADDR=y
Index: u-boot-mox-2018-11-rc1/u-boot/include/configs/turris_mox.h
===================================================================
--- u-boot-mox-2018-11-rc1.orig/u-boot/include/configs/turris_mox.h
+++ u-boot-mox-2018-11-rc1/u-boot/include/configs/turris_mox.h
@@ -38,7 +38,8 @@
 /*
  * Size of malloc() pool
  */
-#define CONFIG_SYS_MALLOC_LEN	(4 << 20) /* 4MiB for malloc() */
+#define CONFIG_SYS_BOOTM_LEN (32<<20)
+#define CONFIG_SYS_MALLOC_LEN	(64<<20) /* 64MiB for malloc() */
 
 /*
  * Other required minimal configurations
@@ -101,6 +102,10 @@
 	"fdt_addr_r=0x4f00000\0"	\
 	"kernel_addr_r=0x5000000\0"	\
 	"ramdisk_addr_r=0x8000000\0"	\
+	"mox_distro_bootcmd=sf probe; sf read ${fdt_addr_r} 0x7f0000 0x10000; run distro_bootcmd\0"	\
+	"rescue_args=console=ttyMV0,115200 earlycon=ar3700_uart,0xd0012000\0"	\
+	"rescue_bootcmd=gpio set GPIO221; setenv bootargs $rescue_args; sf probe; sf read $kernel_addr_r 0x190000; lzmadec $kernel_addr_r $ramdisk_addr_r; bootm $ramdisk_addr_r\0"	\
+	"mox_boot=gpio clear GPIO221; if gpio input GPIO220 && sleep 1 && gpio input GPIO220; then run rescue_bootcmd; else run mox_distro_bootcmd; fi\0"	\
 	BOOTENV
 
 #endif /* _CONFIG_TURRIS_MOX_H */
Index: u-boot-mox-2018-11-rc1/u-boot/arch/arm/dts/armada-3720-turris-mox.dts
===================================================================
--- u-boot-mox-2018-11-rc1.orig/u-boot/arch/arm/dts/armada-3720-turris-mox.dts
+++ u-boot-mox-2018-11-rc1/u-boot/arch/arm/dts/armada-3720-turris-mox.dts
@@ -118,6 +118,31 @@
 		reg = <0>;
 		spi-max-frequency = <20000000>;
 		m25p,fast-read;
+
+		partition@0 {
+			label = "secure-firmware";
+			reg = <0x0 0x20000>;
+		};
+
+		partition@20000 {
+			label = "u-boot";
+			reg = <0x20000 0x160000>;
+		};
+
+		partition@180000 {
+			label = "u-boot-env";
+			reg = <0x180000 0x10000>;
+		};
+
+		partition@190000 {
+			label = "Rescue system";
+			reg = <0x190000 0x660000>;
+		};
+
+		partition@7f0000 {
+			label = "dtb";
+			reg = <0x7f0000 0x10000>;
+		};
 	};
 
 	moxtet@1 {
