Index: dovecot-2.2.33.2/src/lib/byteorder.h
===================================================================
--- dovecot-2.2.33.2.orig/src/lib/byteorder.h
+++ dovecot-2.2.33.2/src/lib/byteorder.h
@@ -23,6 +23,11 @@
 #ifndef BYTEORDER_H
 #define BYTEORDER_H
 
+#undef bswap_8
+#undef bswap_16
+#undef bswap_32
+#undef bswap_64
+
 /*
  * These prototypes exist to catch bugs in the code generating macros below.
  */
