--- a/Makefile.common
+++ b/Makefile.common
@@ -77,15 +77,15 @@ SENDMAILOBJS = $(addprefix $(OUTDIR)/, $
 
 
 $(OUTDIR)/%.o: %.cpp
-	${CC} ${CPPFLAGS} ${CFLAGS} -o $(OUTDIR)/$*.o -c $*.cpp
+	${CXX} ${CPPFLAGS} ${CFLAGS} -o $(OUTDIR)/$*.o -c $*.cpp
 
 all: ${OUTDIR} ${MKMACHDEPINC} ${MAILSVR} ${CRTLCLNT} ${XMCRYPT} ${MKUSERS} ${SENDMAIL}
 
 ${OUTDIR}:
 	@mkdir ${OUTDIR}
 
-${MKMACHDEPINC}: ${MKMACHDEP}
-	./${MKMACHDEP} > ${MKMACHDEPINC}
+#${MKMACHDEPINC}: ${MKMACHDEP}
+#	./${MKMACHDEP} > ${MKMACHDEPINC}
 
 ${MKMACHDEP}: ${MKMACHDEPOBJS}
 	${LD} -o ${MKMACHDEP} ${MKMACHDEPOBJS} ${LDFLAGS}
