--- /dev/null
+++ b/SysMachine.h
@@ -0,0 +1,30 @@
+#ifndef _MACHDEFS_H
+#define _MACHDEFS_H
+
+
+#include <endian.h>
+
+#if __BYTE_ORDER == __LITTLE_ENDIAN
+#undef MACH_BIG_ENDIAN_WORDS
+#undef MACH_BIG_ENDIAN_BITFIELD
+#elif __BYTE_ORDER == __BIG_ENDIAN
+#define MACH_BIG_ENDIAN_WORDS
+#define MACH_BIG_ENDIAN_BITFIELD
+#endif
+
+typedef signed char MachInt8;
+typedef unsigned char MachUInt8;
+#define MACH_TYPE_8BIT char
+
+typedef signed short MachInt16;
+typedef unsigned short MachUInt16;
+#define MACH_TYPE_16BIT short
+
+typedef signed int MachInt32;
+typedef unsigned int MachUInt32;
+#define MACH_TYPE_32BIT int
+
+
+
+#endif
+
