Index: lua-http-0.4/http-0.4-0.rockspec
===================================================================
--- lua-http-0.4.orig/http-0.4-0.rockspec
+++ lua-http-0.4/http-0.4-0.rockspec
@@ -16,8 +16,7 @@ dependencies = {
 	"lua >= 5.1";
 	"compat53 >= 0.3"; -- Only if lua < 5.3
 	"bit32"; -- Only if lua == 5.1
-	"cqueues >= 20161214";
-	"luaossl >= 20161208";
+	--"luaossl >= 20161208";
 	"basexx >= 0.2.0";
 	"lpeg";
 	"lpeg_patterns >= 0.5";
