diff -Naru ldns-1.6.17/Makefile.in ldns-1.6.17/Makefile.in
--- ldns-1.6.17/Makefile.in	2016-04-08 15:54:55.640479771 +0200
+++ ldns-1.6.17/Makefile.in	2016-04-08 15:55:13.968590780 +0200
@@ -270,7 +270,7 @@
 test-p5-dns-ldns: $(p5_dns_ldns_dir)/Makefile
 	cd $(p5_dns_ldns_dir); $(MAKE) test
 
-install: install-h install-lib @INSTALL_CONFIG@ install-manpages $(pyldns_inst) $(pyldnsx_inst) @INSTALL_P5_DNS_LDNS@ @INSTALL_DRILL@ @INSTALL_EXAMPLES@
+install: install-h install-lib @INSTALL_CONFIG@ $(pyldns_inst) $(pyldnsx_inst) @INSTALL_P5_DNS_LDNS@ @INSTALL_DRILL@ @INSTALL_EXAMPLES@
 
 uninstall: uninstall-manpages @UNINSTALL_CONFIG@ uninstall-h uninstall-lib $(pyldns_uninst) $(pyldnsx_uninst) @UNINSTALL_P5_DNS_LDNS@ @UNINSTALL_DRILL@ @UNINSTALL_EXAMPLES@
 
@@ -293,12 +293,6 @@
 	rm -f $(DESTDIR)$(mandir)/man1/ldns-config.1
 	test ! -d $(DESTDIR)$(mandir)/man1 || rmdir -p $(DESTDIR)$(mandir)/man1 || echo "ok, dir already gone"
 
-install-manpages: manpages  @INSTALL_CONFIG_MANPAGE@
-	${INSTALL} -d $(DESTDIR)$(mandir)/man3
-	for f in doc/man/man3/*; do \
-		${INSTALL} -c -m 444 $$f $(DESTDIR)$(mandir)/man3/; \
-	done
-
 uninstall-manpages: @UNINSTALL_CONFIG_MANPAGE@
 	for m in `cat $(srcdir)/ldns/*.h | perl $(srcdir)/doc/doxyparse.pl -m $(srcdir)/doc/function_manpages 2>&1 | grep -v ^doxygen | grep -v ^cat` ; do \
 		rm -f $(DESTDIR)$(mandir)/man3/$$m.3 ; done
