--- a/src/Makefile
+++ b/src/Makefile
@@ -31,16 +31,16 @@ MAINOBJ  := if.o interface.o etherStream
 EXAMPLE  := example
 EXOBJ    := if.o interface.o etherStream.o etherStreambuf.o tcStream.o tcStreambuf.o tcRtsDump.o tcRtsDumpbuf.o
 
-CXX        := g++
-CXXFLAGS   := -march=i386 -O2
-LDFLAGS    := --strip-unneeded
-LIBS       := 
-LDSTATIC   := -L. -static-libgcc
-STATICLIBS := stdc++ m
-COUNTRY    := "C"
-LANG       := "C"
-LANGUAGE   := "C"
-LC_ALL     := "C"
+#CXX        := g++
+#CXXFLAGS   := -march=i386 -O2
+#LDFLAGS    := --strip-unneeded
+#LIBS       := 
+#LDSTATIC   := -L. -static-libgcc
+#STATICLIBS := stdc++ m
+#COUNTRY    := "C"
+#LANG       := "C"
+#LANGUAGE   := "C"
+#LC_ALL     := "C"
 
 default: install
 
