Index: x265_2.8/source/common/param.cpp
===================================================================
--- x265_2.8.orig/source/common/param.cpp
+++ x265_2.8/source/common/param.cpp
@@ -633,7 +633,7 @@ int x265_param_parse(x265_param* p, cons
         if (bValueWasNull)
             p->cpuid = atobool(value);
         else
-            p->cpuid = parseCpuName(value, bError);
+            p->cpuid = parseCpuName(value, bError, false);
 #endif
     }
     OPT("fps")
