diff --git a/foris_plugins/ssbackups/templates/ssbackups/ssbackups.html.j2 b/foris_plugins/ssbackups/templates/ssbackups/ssbackups.html.j2
index d4e0e8de73f64e00ef7f61b96ad04753bf94fa60..615829c3ea5b80a1e56c2b438c4af4ecfdee53d6 100644
--- a/foris_plugins/ssbackups/templates/ssbackups/ssbackups.html.j2
+++ b/foris_plugins/ssbackups/templates/ssbackups/ssbackups.html.j2
@@ -20,7 +20,7 @@
     </form>
     <p>{% trans %}Cloud backups are created automatically every night as well as manually. The service stores up to seven backups, additional backups are gradually overwritten unless the user selects Keep. Once the backup is kept, it will remain stored on a remote server until deleted. {% endtrans %}</p>
     <div class="message warning">
-        {% trans project_url="https://project.turris.cz", data_collect_url=url("config_page", page_name="data-collection") %}To be able to use cloud backups, please make sure that you are registered at the <a href='{{ project_url }}'>Project Turris portal</a>. You can do that on the <a href='{{ data_collect_url }}'>Data Collect tab</a>.{% endtrans %}
+        {% trans project_url="https://project.turris.cz", data_collect_url=url("config_page", page_name="data_collect") %}To be able to use cloud backups, please make sure that you are registered at the <a href='{{ project_url }}'>Project Turris portal</a>. You can do that on the <a href='{{ data_collect_url }}'>Data Collect tab</a>.{% endtrans %}
       ({% trans %}Note that you don't need to enable data collection after the registration.{% endtrans %})
     </div>
     <h3>{% trans %}Remote backups{% endtrans %}</h3>
