diff --git a/foris_controller_backends/openvpn_client/__init__.py b/foris_controller_backends/openvpn_client/__init__.py
index 7be456a3b892fe3e6379e3a763250a281192989a..20a0f1df56f3ee38f56db41a2f3a94179285fa19 100644
--- a/foris_controller_backends/openvpn_client/__init__.py
+++ b/foris_controller_backends/openvpn_client/__init__.py
@@ -1,6 +1,6 @@
 #
 # foris-controller-openvpn_client-module
-# Copyright (C) 2019 CZ.NIC, z.s.p.o. (http://www.nic.cz/)
+# Copyright (C) 2019-2020 CZ.NIC, z.s.p.o. (http://www.nic.cz/)
 #
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -88,6 +88,8 @@ class OpenVpnClientUci:
             backend.set_option("openvpn", id, "enabled", store_bool(True))
             backend.set_option("openvpn", id, "_client_foris", store_bool(True))
             backend.set_option("openvpn", id, "config", str(file_path))
+            backend.set_option("openvpn", id, "dev", f"vpn_{id}")
+            backend.add_to_list("firewall", "turris_vpn_client", "device", [ f"vpn_{id}" ])
 
         with OpenwrtServices() as services:
             MaintainCommands().restart_network()
@@ -126,6 +128,7 @@ class OpenVpnClientUci:
                 return False
 
             backend.del_section("openvpn", id)
+            backend.del_from_list("firewall", "turris_vpn_client", "device", [ f"vpn_{id}" ] )
 
             file_path = pathlib.Path("/etc/openvpn/foris") / f"{id}.conf"
             BaseFile().delete_file(str(file_path))
diff --git a/tests/uci_configs/firewall b/tests/uci_configs/firewall
new file mode 100644
index 0000000000000000000000000000000000000000..8acf19fff0099f917829d52e9c801c08c23863d2
--- /dev/null
+++ b/tests/uci_configs/firewall
@@ -0,0 +1,11 @@
+config zone 'turris_vpn_client'
+        option name 'tr_vpn_cl'
+        option input 'REJECT'
+        option output 'ACCEPT'
+        option forward 'REJECT'
+        option masq '1'
+
+config forwarding 'turris_vpn_client_forward'
+        option src 'lan'
+        option dest 'tr_vpn_cl'
+
diff --git a/uci-defaults/vpn-client-network b/uci-defaults/vpn-client-network
new file mode 100644
index 0000000000000000000000000000000000000000..89bdddde5ade139b7646f3e150a3e896505c778a
--- /dev/null
+++ b/uci-defaults/vpn-client-network
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+if ! uci -q get firewall.turris_openvpn_client > /dev/null; then
+uci batch << EOF
+set firewall.turris_vpn_client=zone
+set firewall.turris_vpn_client.name='tr_vpn_cl'
+set firewall.turris_vpn_client.input='REJECT'
+set firewall.turris_vpn_client.output='ACCEPT'
+set firewall.turris_vpn_client.forward='REJECT'
+set firewall.turris_vpn_client.masq='1'
+set firewall.turris_vpn_client_forward='forwarding'
+set firewall.turris_vpn_client_forward.src='lan'
+set firewall.turris_vpn_client_forward.dest='tr_vpn_cl'
+commit firewall
+EOF
+fi
