Index: wrapsix-0.2.1/src/wrapper.c
===================================================================
--- wrapsix-0.2.1.orig/src/wrapper.c
+++ wrapsix-0.2.1/src/wrapper.c
@@ -22,7 +22,7 @@
 #include <linux/ethtool.h>	/* struct ethtool_value */
-#include <linux/if_ether.h>	/* ETH_P_ALL */
+#define _NETINET_IF_ETHER_H
 #include <linux/sockios.h>	/* SIOCETHTOOL */
 #include <net/ethernet.h>	/* ETHERTYPE_* */
 #include <net/if.h>		/* struct ifreq */
 #include <netinet/in.h>		/* htons */
 #include <netpacket/packet.h>	/* struct packet_mreq, struct sockaddr_ll */
