From c299d82d3b6c2e8f586a53aa16e80938e52778a3 Mon Sep 17 00:00:00 2001
From: Michal Vasilek <michal.vasilek@nic.cz>
Date: Fri, 22 Oct 2021 01:38:49 +0200
Subject: [PATCH] uncomment auth module import

---
 doc/config/conf.d/auth.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/config/conf.d/auth.conf b/doc/config/conf.d/auth.conf
index fa4c86d..6039478 100644
--- a/doc/config/conf.d/auth.conf
+++ b/doc/config/conf.d/auth.conf
@@ -7,7 +7,7 @@
 ## for more info and many more options.
 ##
 
-#server.modules += ( "mod_auth" )
+server.modules += ( "mod_auth" )
 
 #server.modules += ( "mod_authn_file" )
 #auth.backend                 = "plain"
-- 
2.33.1

