--- a/configure.ac
+++ b/configure.ac
@@ -763,7 +763,6 @@ libs/gst/net/Makefile
 plugins/Makefile
 plugins/indexers/Makefile
 plugins/elements/Makefile
-po/Makefile.in
 tests/Makefile
 tests/benchmarks/Makefile
 tests/check/Makefile
--- a/Makefile.am
+++ b/Makefile.am
@@ -8,14 +8,12 @@ aclocal_DATA = gst-element-check-@GST_MA
 
 SUBDIRS = pkgconfig \
 	gst libs plugins tools \
-	po \
 	m4 \
 	common
 
 # These are all the possible subdirs
 DIST_SUBDIRS = pkgconfig \
 	gst libs plugins tools \
-	po \
 	m4 \
 	common
 
--- a/Makefile.in
+++ b/Makefile.in
@@ -469,7 +469,6 @@ aclocaldir = $(datadir)/aclocal
 aclocal_DATA = gst-element-check-@GST_MAJORMINOR@.m4
 SUBDIRS = pkgconfig \
 	gst libs plugins tools \
-	po \
 	m4 \
 	common
 
@@ -477,7 +476,6 @@ SUBDIRS = pkgconfig \
 # These are all the possible subdirs
 DIST_SUBDIRS = pkgconfig \
 	gst libs plugins tools \
-	po \
 	m4 \
 	common
 
