Index: foris-plugins-distutils/setup.py
===================================================================
--- foris-plugins-distutils.orig/setup.py
+++ foris-plugins-distutils/setup.py
@@ -10,8 +10,9 @@ setup(
     author_email="stepan.henek@nic.cz",
     url="https://gitlab.labs.nic.cz/turris/foris-plugins-distutils/",
     license="GPL-3.0",
+    setup_requires=['babel','libsass','jinja2'],
     requires=[],
-    install_requires=["babel", "jinja2", "libsass"],
+    #install_requires=["babel", "jinja2", "libsass"],
     provides=["foris_plugins_distutils"],
     packages=["foris_plugins_distutils"],
     entry_points={
