--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -713,6 +713,8 @@ Programs/_testembed: Programs/_testembed
 ############################################################################
 # Importlib
 
+FREEZE_IMPORTLIB ?= ./Programs/_freeze_importlib
+
 Programs/_freeze_importlib.o: Programs/_freeze_importlib.c Makefile
 
 Programs/_freeze_importlib: Programs/_freeze_importlib.o $(LIBRARY_OBJS_OMIT_FROZEN)
@@ -720,13 +722,13 @@ Programs/_freeze_importlib: Programs/_fr
 
 Python/importlib_external.h: $(srcdir)/Lib/importlib/_bootstrap_external.py Programs/_freeze_importlib
 	if test "$(cross_compiling)" != "yes"; then \
-	    ./Programs/_freeze_importlib \
+	    $(FREEZE_IMPORTLIB) \
 	        $(srcdir)/Lib/importlib/_bootstrap_external.py Python/importlib_external.h; \
 	fi
 
 Python/importlib.h: $(srcdir)/Lib/importlib/_bootstrap.py Programs/_freeze_importlib
 	if test "$(cross_compiling)" != "yes"; then \
-	    ./Programs/_freeze_importlib \
+	    $(FREEZE_IMPORTLIB) \
 	        $(srcdir)/Lib/importlib/_bootstrap.py Python/importlib.h; \
 	fi
 
