--- a/configure
+++ b/configure
@@ -5277,7 +5277,7 @@ cat >> conftest.c <<EOF
 # elif defined(__or1k__)
         or1k-linux-gnu
 # elif defined(__powerpc__) && defined(__SPE__)
-        powerpc-linux-gnuspe
+        powerpc-linux-gnuspev1
 # elif defined(__powerpc64__)
 #  if defined(__LITTLE_ENDIAN__)
         powerpc64le-linux-gnu
--- a/configure.ac
+++ b/configure.ac
@@ -819,7 +819,7 @@ cat >> conftest.c <<EOF
 # elif defined(__or1k__)
         or1k-linux-gnu
 # elif defined(__powerpc__) && defined(__SPE__)
-        powerpc-linux-gnuspe
+        powerpc-linux-gnuspev1
 # elif defined(__powerpc64__)
 #  if defined(__LITTLE_ENDIAN__)
         powerpc64le-linux-gnu
