--- a/Makefile.am
+++ b/Makefile.am
@@ -12,14 +12,12 @@ SUBDIRS = 			\
 	gst sys $(SUBDIRS_EXT) 	\
 	tools 			\
 	tests 			\
-	docs			\
 	po 			\
 	common 			\
 	m4
 
 DIST_SUBDIRS = 			\
 	pkgconfig		\
-	docs			\
 	gst-libs		\
 	gst sys ext		\
 	tools 			\
--- a/Makefile.in
+++ b/Makefile.in
@@ -464,14 +464,12 @@ SUBDIRS = \
 	gst sys $(SUBDIRS_EXT) 	\
 	tools 			\
 	tests 			\
-	docs			\
 	po 			\
 	common 			\
 	m4
 
 DIST_SUBDIRS = \
 	pkgconfig		\
-	docs			\
 	gst-libs		\
 	gst sys ext		\
 	tools 			\
