diff -ur a/lib/readline/display.c b/lib/readline/display.c
--- a/lib/readline/display.c	2014-09-25 13:08:31.323147031 +0200
+++ b/lib/readline/display.c	2014-09-25 13:08:31.353147030 +0200
@@ -2677,7 +2677,8 @@
 {
   if (_rl_echoing_p)
     {
-      _rl_move_vert (_rl_vis_botlin);
+      if (_rl_vis_botlin > 0)	/* minor optimization plus bug fix */
+	_rl_move_vert (_rl_vis_botlin);
       _rl_vis_botlin = 0;
       fflush (rl_outstream);
       rl_restart_output (1, 0);
diff -ur a/patchlevel.h b/patchlevel.h
--- a/patchlevel.h	2014-09-25 13:08:31.309813698 +0200
+++ b/patchlevel.h	2014-09-25 13:08:31.353147030 +0200
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 10
+#define PATCHLEVEL 11
 
 #endif /* _PATCHLEVEL_H_ */
