diff -ur a/arrayfunc.c b/arrayfunc.c
--- a/arrayfunc.c	2014-09-25 13:08:32.126480355 +0200
+++ b/arrayfunc.c	2014-09-25 13:08:32.166480355 +0200
@@ -179,6 +179,7 @@
     array_insert (array_cell (entry), ind, newval);
   FREE (newval);
 
+  VUNSETATTR (entry, att_invisible);	/* no longer invisible */
   return (entry);
 }
 
diff -ur a/patchlevel.h b/patchlevel.h
--- a/patchlevel.h	2014-09-25 13:08:32.123147021 +0200
+++ b/patchlevel.h	2014-09-25 13:08:32.169813688 +0200
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 17
+#define PATCHLEVEL 18
 
 #endif /* _PATCHLEVEL_H_ */
