Index: mariadb-10.4.6/scripts/mysql_install_db.sh
===================================================================
--- mariadb-10.4.6.orig/scripts/mysql_install_db.sh
+++ mariadb-10.4.6/scripts/mysql_install_db.sh
@@ -467,21 +467,24 @@ done
 
 if test -n "$user"
 then
-  chown $user "$pamtooldir/auth_pam_tool_dir"
-  if test $? -ne 0
+  if test -d "$pamtooldir/auth_pam_tool_dir"
   then
-      echo "Cannot change ownership of the '$pamtooldir/auth_pam_tool_dir' directory"
-      echo " to the '$user' user. Check that you have the necessary permissions and try again."
-      exit 1
-  fi
-  if test -z "$srcdir"
-  then
-    chown 0 "$pamtooldir/auth_pam_tool_dir/auth_pam_tool"
+    chown $user "$pamtooldir/auth_pam_tool_dir"
     if test $? -ne 0
     then
-        echo "Couldn't set an owner to '$pamtooldir/auth_pam_tool_dir/auth_pam_tool'."
-        echo " It must be root, the PAM authentication plugin doesn't work otherwise.."
-        echo
+	echo "Cannot change ownership of the '$pamtooldir/auth_pam_tool_dir' directory"
+	echo " to the '$user' user. Check that you have the necessary permissions and try again."
+	exit 1
+    fi
+    if test -z "$srcdir"
+    then
+      chown 0 "$pamtooldir/auth_pam_tool_dir/auth_pam_tool"
+      if test $? -ne 0
+      then
+	  echo "Couldn't set an owner to '$pamtooldir/auth_pam_tool_dir/auth_pam_tool'."
+	  echo " It must be root, the PAM authentication plugin doesn't work otherwise.."
+	  echo
+      fi
     fi
   fi
   args="$args --user=$user"
