From 40aa3daf7522c997b1d56aae3dfe5de1b0de9294 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Karel=20Ko=C4=8D=C3=AD?= <karel.koci@nic.cz>
Date: Mon, 25 Mar 2019 07:29:33 +0100
Subject: [PATCH] Fix entry point

---
 svupdater/__init__.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/svupdater/__init__.py b/svupdater/__init__.py
index 53e2508..f51db4f 100644
--- a/svupdater/__init__.py
+++ b/svupdater/__init__.py
@@ -23,6 +23,7 @@
 # NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
 # EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 from . import autorun, const
+from .__main__ import main
 from .utils import check_exclusive_lock as _check_exclusive_lock
 from .utils import daemonize as _daemonize
 from ._pidlock import pid_locked as _pid_locked
-- 
2.21.0

