From 6a38f3d8f6f7df27b5a14533628187a2a3962b92 Mon Sep 17 00:00:00 2001
From: Stepan Henek <stepan.henek@nic.cz>
Date: Tue, 19 Mar 2019 15:23:36 +0100
Subject: [PATCH] disable WS filtering when controller-id is not set

---
 foris/static/js/foris.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/foris/static/js/foris.js b/foris/static/js/foris.js
index 83117b6..4b5450d 100644
--- a/foris/static/js/foris.js
+++ b/foris/static/js/foris.js
@@ -134,7 +134,7 @@ Foris.applySVGFallback = function() {
 Foris.WsHandlers = {};
 
 Foris.addWsHanlder = (module, handler, controller_id) => {
-    controller_id = controller_id || $('meta[name=controller-id]').attr("content");
+    controller_id = controller_id || $('meta[name=controller-id]').attr("content") || '+';
     let moduleHandler = Foris.WsHandlers[module] || {};
     let handlers = moduleHandler[controller_id] || [];
     handlers.push(handler);
-- 
2.20.1

