commit 63102336cc44fe523e4e72936eddfead3fbdf5ab
Author: Michal Hrusecky <michal.hrusecky@nic.cz>
Date:   Thu Apr 25 10:54:38 2019 +0200

    guide: Mention security research in finished dialog

diff --git a/foris/config_handlers/misc.py b/foris/config_handlers/misc.py
index 91fbf87..bed48c5 100644
--- a/foris/config_handlers/misc.py
+++ b/foris/config_handlers/misc.py
@@ -316,14 +316,7 @@ class GuideFinishedHandler(BaseConfigHandler):
 
     def get_form(self):
         finished_form = fapi.ForisForm("guide_finished", {})
-        finished_form.add_section(
-            name="guide_finished",  title=_(self.userfriendly_title),
-            description=_(
-                "Congratulations you've successfully reached the end of this guide. "
-                "Once you leave this guide you'll be granted access to the "
-                "full configuration interface of this device."
-            )
-        )
+        finished_form.add_section(name="guide_finished", title=_(self.userfriendly_title))
 
         def guide_finished_cb(data):
             res = current_state.backend.perform("web", "update_guide", {
diff --git a/foris/templates/config/finished.html.j2 b/foris/templates/config/finished.html.j2
index 95bea32..0b5bbf5 100644
--- a/foris/templates/config/finished.html.j2
+++ b/foris/templates/config/finished.html.j2
@@ -5,7 +5,22 @@
 <div id="page-config" class="config-page">
 {% endif %}
     <form id="main-form" class="config-form" action="{{ request.fullpath }}" method="post" autocomplete="off" novalidate>
-        <p class="config-description">{{ description|safe }}</p>
+        <p class="config-description">
+            {% trans -%}
+            Congratulations you've successfully reached the end of this guide.
+            Once you leave this guide you'll be granted access to the
+            full configuration interface of this device.
+            {%- endtrans %}
+        </p>
+        <p>
+            {% trans -%}
+            To further <strong>improve your security</strong> consider enabling data
+            collect (start by selecting it in updater tab). This will allow
+            you to be part of our security research to discover new
+            attackers and it will also give you access to dynamic updates
+            to your firewall to block all already known attackers.
+            {%- endtrans %}
+        </p>
         {% include '_messages.html.j2' %}
         <input type="hidden" name="csrf_token" value="{{ get_csrf_token() }}">
         <div id="{{ 'form-%s-buttons' % form.name }}" class="form-buttons">
